site/src/pages/WorkspacesPage/WorkspacesPage.tsx

```diff
@@ -1,20 +1,99 @@
+import Button from "@material-ui/core/Button"
+import Link from "@material-ui/core/Link"
+import Menu from "@material-ui/core/Menu"
+import MenuItem from "@material-ui/core/MenuItem"
+import { makeStyles } from "@material-ui/core/styles"
+import TextField from "@material-ui/core/TextField"
+import AddCircleOutline from "@material-ui/icons/AddCircleOutline"
 import { useMachine } from "@xstate/react"
-import { FC } from "react"
+import { FormikContextType, FormikErrors, useFormik } from "formik"
+import { FC, useState } from "react"
+import { Link as RouterLink } from "react-router-dom"
+import { Margins } from "../../components/Margins/Margins"
+import { Stack } from "../../components/Stack/Stack"
+import { getFormHelpers, onChangeTrimmed } from "../../util/formUtils"
 import { workspacesMachine } from "../../xServices/workspaces/workspacesXService"
-import { WorkspacesPageView } from "./WorkspacesPageView"
+import { Language, WorkspacesPageView } from "./WorkspacesPageView"
+
+interface FilterFormValues {
+  query: string
+}
+
+export type FilterFormErrors = FormikErrors<FilterFormValues>
 
 const WorkspacesPage: FC = () => {
-  const [workspacesState] = useMachine(workspacesMachine)
+  const styles = useStyles()
+  const [workspacesState, send] = useMachine(workspacesMachine)
+
+  const form: FormikContextType<FilterFormValues> = useFormik<FilterFormValues>({
+    initialValues: { query: workspacesState.context.filter || "" },
+    onSubmit: (data) => {
+      send({
+        type: "SET_FILTER",
+        query: data.query,
+      })
+    },
+  })
```

Comment:
```suggestion
const form = useFormik<FilterFormValues>({
    initialValues: {
      query: workspacesState.context.filter || "",
    },
    onSubmit: (values) => {
      send({
        type: "SET_FILTER",
        query: values.query,
      })
    },
  })
```

Small cleanup here for preferred TS readability and conventions